NameMangling.md

```diff
@@ -0,0 +1,37 @@
+# Name mangling
+
+Linker will mangle names internally,
+but we're hoping to keep the mangling details
+out of the public api.
+
+See also renaming host visible names in [Visiblity](./Visiblity.md).
+
+(TBD)
+
+## Mangling
```

Comment:
Interestingly, this standard WGSL mangling scheme might need to be extended to support
- WGSL modules that are outside of the package root (not sure if they should be supported)
- Generics 😎

Reply:
Adding comment re: generics.

agreed, re limiting imports to within the package_root. If users want to import from an uncle dir, they can move the package_root up one dir.